src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al

```diff
@@ -0,0 +1,509 @@
+        NoSeriesLine."Line No." := GetNoSeriesLineNo(NoSeriesGenerated."Series Code");
+        NoSeriesLine."Starting No." := NoSeriesGenerated."Starting No.";
+        NoSeriesLine."Ending No." := NoSeriesGenerated."Ending No.";
+        NoSeriesLine."Warning No." := NoSeriesGenerated."Warning No.";
+        NoSeriesLine."Increment-by No." := NoSeriesGenerated."Increment-by No.";
+        //TODO: Check if we need to add more fields here, like "Allow Gaps in Nos.", "Sequence Name" etc.
+        if not NoSeriesLine.Insert(true) then
+            NoSeriesLine.Modify(true);
+    end;
+
+    local procedure GetNoSeriesLineNo(SeriesCode: Code[20]): Integer
+    var
+        NoSeriesLine: Record "No. Series Line";
+        NoSeries: Codeunit "No. Series";
+    begin
+        if not NoSeries.GetNoSeriesLine(NoSeriesLine, SeriesCode, 0D, true) then
+            exit(1000);
+
+        exit(NoSeriesLine."Line No."); // TODO: Check if we need to update existing no series line, or add a new one, e.g. if user requested to create no. series for the new year
+    end;
+
+    local procedure ApplyNoSeriesToSetup(var NoSeriesGenerated: Record "No. Series Proposal Line")
+    var
+        RecRef: RecordRef;
+        FieldRef: FieldRef;
+    begin
+        RecRef.OPEN(NoSeriesGenerated."Setup Table No.");
+        if not RecRef.FindFirst() then
+            exit;
+
+        FieldRef := RecRef.Field(NoSeriesGenerated."Setup Field No.");
+        FieldRef.Validate(NoSeriesGenerated."Series Code");
+        RecRef.Modify(true);
+    end;
+
+    [NonDebuggable]
+    local procedure GetNoSeriesGenerationSystemPrompt() SystemPrompt: Text
+    var
+        NoSeriesCopilotSetup: Record "No. Series Copilot Setup";
+    begin
+        // This is a temporary solution to get the system prompt. The system prompt should be retrieved from the Azure Key Vault.
+        // TODO: Retrieve the system prompt from the Azure Key Vault, when passed all tests.
+        NoSeriesCopilotSetup.Get();
+        SystemPrompt := NoSeriesCopilotSetup.GetNoSeriesGenerationSystemPromptFromIsolatedStorage().Replace(DateSpecificPlaceholderLbl, Format(Today(), 0, 4));
+    end;
+
+    [NonDebuggable]
+    local procedure GetToolsSystemPrompt() SystemPrompt: Text
+    var
+        NoSeriesCopilotSetup: Record "No. Series Copilot Setup";
+    begin
+        // This is a temporary solution to get the system prompt. The system prompt should be retrieved from the Azure Key Vault.
+        // TODO: Retrieve the system prompt from the Azure Key Vault, when passed all tests.
+        NoSeriesCopilotSetup.Get();
+        SystemPrompt := NoSeriesCopilotSetup.GetToolsSystemPromptFromIsolatedStorage().Replace(DateSpecificPlaceholderLbl, Format(Today(), 0, 4));
+    end;
+
+    [NonDebuggable]
+    local procedure GenerateNoSeries(SystemPromptTxt: SecretText; InputText: Text): Text
+    var
+        AzureOpenAI: Codeunit "Azure OpenAi";
+        AOAIDeployments: Codeunit "AOAI Deployments";
```

Comment:
Unused. You should enable the CodeCops to get all of minor issues sorted out. Won't mention further of those.